ibis/pandas/execution/tests/test_window.py

```diff
@@ -703,3 +761,35 @@ def get_aggcontext_custom(
     expected = pd.Series([4.0, 10.0, 5.0])
 
     tm.assert_series_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    'table_name,column,order_by,group_by,bucket,expected',
+    [
+        ('salary_1', 'salary', ['salary'], None, 4, [0, 0, 1, 1, 2, 3],),
+        (
+            'salary_2',
+            'amount',
+            ['year', 'amount'],
+            ['year'],
+            3,
+            [0, 0, 1, 1, 2, 0, 0, 1, 1, 2],
+        ),
+    ],
+)
+def test_ntile(
+    con_salary,
+    table_name: str,
+    column: str,
+    order_by: List[str],
+    group_by: List[str],
+    bucket: int,
+    expected: List[int],
+):
+    """Test ntile operation."""
+    t = con_salary.table(table_name)
+
+    win = ibis.window(following=0, group_by=group_by, order_by=order_by)
+    expr = t.mutate(val=t[column].ntile(buckets=bucket).over(win))
+    result = expr.execute().sort_values(order_by)
+    assert result.val.tolist() == expected
```

Comment:
pls do not do this, *always* use ``tm.assert_frame_equal`` and construc the expected frame. a list of things is NOT a dataframe and doesn't not validate types, nor metadata (columns)